cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/RunRecordCorrectorServiceTest.java

```diff
@@ -129,6 +129,8 @@ public ProgramRunStatus call() throws Exception {
     // Start the RunRecordCorrectorService, which will fix the run record
     new LocalRunRecordCorrectorService(store, programStateWriter, programLifecycleService, runtimeService).startUp();
 
+    // Sleep to make sure the run record correction notification is processed
+    Thread.sleep(3000);
```

Comment:
Let's use `Tasks.waitFor()` with 30 seconds, so that test does not become flaky.